### PR TITLE
Make `localwhitelist` check if caller will still be able to use bot after changes

### LINKED ISSUE #3222: [Core] Users may accidentally lock guild owner and themselves with `[p]localwhitelist add`
# Command bugs

#### Command name

`[p]localwhitelist add`

#### What cog is this command from?

Core

#### What were you expecting to happen?

I was expecting to get an error saying that I can't lock out myself (and guild owner) from using the bot.

#### What actually happened?

I locked myself and guild owner out.

#### How can we reproduce this issue?

1. Have permissions to use `[p]localwhitelist` (e.g. have admin role), but don't be bot owner.
2. Use `[p]localwhitelist clear`
3. Use `[p]localwhitelist add SOMEONE_ELSES_USER_ID`

---

This doesn't seem like a trivial task but it would surely be great to have it improved.

### PR DESCRIPTION
**Pull request by Drapersniper**
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature

### Description of the changes
fixes #3222

### CODE REVIEW

**Comment by Jackenmen** [comment]
Adding guild owner is actually unnecessary - the linked issue description is slightly outdated and we always return `True` for guild owner already:
https://github.com/Cog-Creators/Red-DiscordBot/blob/2c313594dacf298e47e3e5ec339f2af2c545b9b2/redbot/core/bot.py#L383-L385

*(I just happened to look at issue tracker :P)*

**Comment by Drapersniper** [comment]
> Adding guild owner is actually unnecessary - the linked issue description is slightly outdated and we always return `True` for guild owner already:
> https://github.com/Cog-Creators/Red-DiscordBot/blob/2c313594dacf298e47e3e5ec339f2af2c545b9b2/redbot/core/bot.py#L383-L385
> 
> _(I just happened to look at issue tracker :P)_

yeah i saw this after i added but forgot to remove it, thanks for spotting it

**Review by Jackenmen — CHANGES_REQUESTED** [review]
Similar check is needed `[p]localwhitelist remove` as well - I can add myself, then someone else and later remove myself and then I will lose my access to the bot.

Also, I was thinking that the error message in `[p]localwhitelist add` could mention that to prevent access loss, you should add yourself to the whitelist first.

**Review by Jackenmen — CHANGES_REQUESTED** [review]
The comment I made in Discord that would be good to address in this PR:
These checks are not needed for guild owners and bot owners as they're not affected by the local whitelist and it may make it unnecessarily harder to add/remove some role on the whitelist.

File: redbot/core/core_commands.py [hunk 1]
```diff
@@ -1946,8 +1946,20 @@ async def localwhitelist_add(
             return
 
         names = [getattr(u_or_r, "name", u_or_r) for u_or_r in users_or_roles]
-        uids = [getattr(u_or_r, "id", u_or_r) for u_or_r in users_or_roles]
-        await self.bot._whiteblacklist_cache.add_to_whitelist(ctx.guild, uids)
+        uids = {getattr(u_or_r, "id", u_or_r) for u_or_r in users_or_roles}
+        if not (ctx.guild.owner == ctx.author or await self.bot.is_owner(ctx.author)):
+            current_whitelist = await self.bot._whiteblacklist_cache.get_whitelist(ctx.guild)
+            theoretical_whitelist = current_whitelist + uids
```
Comment (Jackenmen) [inline redbot/core/core_commands.py]: You're concatenating list and set here which raises an error.

**Review by Jackenmen — CHANGES_REQUESTED** [review]

**Review by Jackenmen — APPROVED** [review]
Looks good to me.

**Pull request merged**